src/Loggers/Loggers.cpp

```diff
@@ -34,6 +34,16 @@ static std::string createDirectory(const std::string & file)
     return path;
 }
 
+static std::string renderFileNameTemplate(time_t now, const std::string & file_path)
+{
+    fs::path path{file_path};
+    std::tm buf;
+    localtime_r(&now, &buf);
+    std::ostringstream ss; // STYLE_CHECK_ALLOW_STD_STRING_STREAM
+    ss << std::put_time(&buf, file_path.c_str());
+    return path.replace_filename(ss.str());
```

Comment:
This breaks relative path configuration for non-daemon servers. Consider the following example:

```
    <logger>
        <level>trace</level>
        <log>logs/clickhouse-server.log</log>
        <errorlog>logs/clickhouse-server.err.log</errorlog>
        <size>1000M</size>
        <count>10</count>
    </logger>
```

The log path will be modified as `logs/logs/clickhouse-server.log`.

Reply:
That's the exact reason why those format specifiers currently do not apply to the "path" part - only to the file name.

Reply:
@amosbird, `path.replace_filename()` does the trick.
If someone decides another day to apply the specifiers to the full path, they will have to take into consideration this issue about non-daemon servers.

Reply:
> That's the exact reason why those format specifiers currently do not apply to the "path" part - only to the file name.

I'm confused. If that's the expected behavior, why not do the following:

```
    ss << std::put_time(&buf, path.filename().c_str());
```

Reply:
@amosbird, you are right, my code looks like an ugly typo, to say the least. This one is more concise, clear and correct:
```cpp
ss << std::put_time(&buf, path.filename().c_str());
```
It looks like it should be amended.

Reply:
@amosbird, you are right, my code looks like an ugly typo, to say the least. This one is more concise, clear and correct:
```cpp
ss << std::put_time(&buf, path.filename().c_str());
```
It looks like it should be amended.

Reply:
I guess, `replace_filename()` extracts the file name from its argument, otherwise it wouldn't work properly.

Reply:
And thank you for noticing.
I'm thinking of creating a "follow-up" PR to clean up this line of code. Or may be you'd rather amend this code by yourself?

Reply:
> And thank you for noticing.

No problem! 

> I'm thinking of creating a "follow-up" PR to clean up this line of code. 

Perfect. Please go ahead for amending it :)